hpat/hiframes/dataframe_pass.py

```diff
@@ -35,24 +36,36 @@
 from hpat.hiframes.pd_rolling_ext import RollingType
 from hpat.hiframes.aggregate import get_agg_func
 
+import types as pytypes
 
-class DataFramePass(object):
+
+@register_pass(mutates_CFG=True, analysis_only=False)
+class DataFramePass(FunctionPass):
     """Analyze and transform dataframe calls after typing"""
 
-    def __init__(self, func_ir, typingctx, typemap, calltypes):
-        self.func_ir = func_ir
-        self.typingctx = typingctx
-        self.typemap = typemap
-        self.calltypes = calltypes
+    _name = "dataframe_pass"
+
+    def __init__(self):
+        pass
+
+    def run_pass(self, state):
+        return DataFramePassImpl(state).run_pass()
+
+
+class DataFramePassImpl(object):
 
-    def run(self):
-        blocks = self.func_ir.blocks
+    def __init__(self, state):
+        self.state = state
```

Comment:
`self.func_ir, self.typemap, self.typingctx = self`
?
no need to keep new variable in this case